ext/spl/spl_array.stub.php

```diff
@@ -4,39 +4,34 @@
 
 class ArrayObject implements IteratorAggregate, ArrayAccess, Serializable, Countable
 {
-    /** @param array|object $input */
-    public function __construct($input = [], int $flags = 0, string $iterator_class = ArrayIterator::class) {}
+    public function __construct(array|object $input = [], int $flags = 0, string $iterator_class = ArrayIterator::class) {}
 
     /**
-     * @param mixed $index
+     * @param array|string|float|int|bool|null $index
```

Comment:
```suggestion
     * @param string|float|int|bool|null $index
```
Though I'm not sure it makes sense to list them as such. I'd probably either keep this `mixed` or only specify `string|int`. The rest are conversions to string or int.

Reply:
Makes sense, this long list was not a very good idea.Though, I'm wondering why to type hint against `string|int` when the argument is converted to an `int`?

Reply:
Oh, please forget this comment. I thought you're referring to `SplDoublyLinkedList`